### ISSUE TITLE
Cannot read property 'name' of null

### ISSUE THREAD

**Issue by MoOx** [brigand/babel-plugin-flow-react-proptypes#66]
I have the following error with 0.20.0 (which was not present with 0.10.1

```
TypeError: src/static/url-as-html.js: Cannot read property 'name' of null
    at annotate (/.../node_modules/babel-plugin-flow-react-proptypes/lib/index.js:84:26)
    at functionVisitor (/.../node_modules/babel-plugin-flow-react-proptypes/lib/index.js:111:7)
    at PluginPass.FunctionDeclaration (/.../node_modules/babel-plugin-flow-react-proptypes/lib/index.js:178:16)
    at newFn (/.../node_modules/babel-traverse/lib/visitors.js:276:21)
    at NodePath._call (/.../node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (/.../node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (/.../node_modules/babel-traverse/lib/path/context.js:105:12)
    at TraversalContext.visitQueue (/.../node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitSingle (/.../node_modules/babel-traverse/lib/context.js:108:19)
    at TraversalContext.visit (/.../node_modules/babel-traverse/lib/context.js:192:19)
    at Function.traverse.node (/.../node_modules/babel-traverse/lib/index.js:114:17)
```

Here is the file (it's not a react components, but I babelify an entire folder)

```
// @flow

import React from "react"
import ReactDOMserver from "react-dom/server"
import { match, RouterContext as RouterContextProvider } from "react-router"
import { Provider as ReduxContextProvider } from "react-redux"

import DefaultHtml from "../components/Html"
import pathToUri from "../_utils/path-to-uri"
import PhenomicContextProvider from "../components/ContextProvider"
import serialize from "../_utils/serialize"
import minifyCollection from "../loader/minify"

export default function(
  url: string,
  options: PhenomicStaticConfig,
  Html: Function = DefaultHtml
): Promise<string> {

  const {
    baseUrl,
    assetsFiles,
    routes,
    collection,
    metadata,
    store,
  } = options

  const render = ReactDOMserver[
    (!options.clientScripts)
    ? "renderToStaticMarkup"
    : "renderToString"
  ]

  return new Promise((resolve, reject) => {
    try {
      match(
        {
          routes,
          location: url,
          basename: baseUrl.pathname,
        },
        (error, redirectLocation, renderProps) => {
          if (error) {
            return reject(error)
          }
          else if (redirectLocation) {
            // TODO add a redirect page à la "jekyll redirect plugin"
            throw new Error (
              "phenomic (static) doesn't handle redirection yet"
            )
          }
          else if (!renderProps) {
            throw new Error (
              "phenomic (static) doesn't handle page not found yet. " +
              "You are not supposed so see this message because this code is " +
              "not supposed to be executed the way thing are, so this can " +
              "be a react-router issue. Check out opened issue to find a " +
              "workaround: https://github.com/MoOx/phenomic/issues"
            )
          }

          const collectionMin = minifyCollection(collection)

          /* eslint-disable react/no-multi-comp */

          const renderBody = () => render(
            <PhenomicContextProvider
              collection={ collectionMin }
              metadata={ metadata }
            >
              <ReduxContextProvider store={ store }>
                <RouterContextProvider { ...renderProps } />
              </ReduxContextProvider>
            </PhenomicContextProvider>
          )

          const renderScript = () => {
            if (options.clientScripts) {
              const initialState = {
                ...store.getState(),
                // only keep current page as others are not necessary
                pages: {
                  [url]: store.getState().pages[url],
                },
              }
              const script = (
                `window.__COLLECTION__ = ${ serialize(collectionMin) };` +
                `window.__INITIAL_STATE__ = ${ serialize(initialState) }`
              )

              return (
                <script dangerouslySetInnerHTML={{ __html: script }} />
              )
            }

            return null
          }

          // write htmlString as html files
          return resolve(
            // render html document as simple html
            "<!doctype html>" +
            ReactDOMserver.renderToStaticMarkup(
              React.createElement(
                Html,
                {
                  ...assetsFiles && {
                    css: assetsFiles.css
                    ? assetsFiles.css.map(
                      (fileName) => pathToUri(baseUrl.pathname, fileName)
                    )
                    : [],
                    js: options.clientScripts && assetsFiles.js
                    ? assetsFiles.js.map(
                      (fileName) => pathToUri(baseUrl.pathname, fileName)
                    )
                    : [],
                  },
                  renderBody,
                  renderScript,
                }
              )
            )
          )
        }
      )
    }
    catch (err) {
      reject(err)
    }
  })
}

```

**Comment by charlesoconor** [comment]
I'm running into this issue to. I believe it has to do with exporting and anonymous functions. I've just flipped the pluggin of for any file I do this in. It would be great to have another work around.

**Comment by mhaas** [comment]
Can you provide a more minimal example which shows this behavior? Ideally, it would be a test case similar to what we have in `src/tests/`

**Comment by mhaas** [comment]
I can reproduce this (or rather, a similar error) and will fix soon.

**Comment by brigand** [comment]
Fixed in #99, published as 3.1.3.
